test/e2e/proxy_test.go

```diff
@@ -68,7 +70,493 @@ func skipIfProxyDisabled(t *testing.T) {
+	errMsg := "Server LoadBalancer whose externalTrafficPolicy is Cluster should be able to be connected from pod"
+	_, _, err := data.runCommandFromPod(testNamespace, clientCp, busyboxContainerName, []string{"wget", "-O", "-", url, "-T", "1"})
+	require.NoError(t, err, errMsg)
+	_, _, err = data.runCommandFromPod(testNamespace, clientWk, busyboxContainerName, []string{"wget", "-O", "-", url, "-T", "1"})
+	require.NoError(t, err, errMsg)
+}
+
+func testLoadBalancerLocalFromLocal(t *testing.T, data *TestData, url, nodeHostnameCp, nodeHostnameWk string) {
+	errMsg := "Server LoadBalancer whose externalTrafficPolicy is Local should be able to be connected from local"
+	getHostNameUrl := fmt.Sprintf("%s/%s", url, "hostname")
+
+	nodeCp := controlPlaneNodeName()
+	skipIfKubeProxyEnabledOnLinux(t, data, nodeCp)
+	_, output, _, err := RunCommandOnNode(nodeCp, fmt.Sprintf("curl --connect-timeout 1 --retry 5 --retry-connrefused %s", getHostNameUrl))
+	require.NoError(t, err, errMsg)
+	require.Contains(t, output, nodeHostnameCp, fmt.Sprintf("hostname should be %s", nodeHostnameCp))
+
+	nodeWk := workerNodeName(1)
+	skipIfKubeProxyEnabledOnLinux(t, data, nodeWk)
+	_, output, _, err = RunCommandOnNode(nodeWk, fmt.Sprintf("curl --connect-timeout 1 --retry 5 --retry-connrefused %s", getHostNameUrl))
+	require.NoError(t, err, errMsg)
+	require.Contains(t, output, nodeHostnameWk, fmt.Sprintf("hostname should be %s", nodeHostnameWk))
+}
+
+func testLoadBalancerLocalFromPod(t *testing.T, data *TestData, url, clientCp, clientWk, clientCpIP, clientWkIP, testPodHostnameCp, testPodHostnameWk string) {
+	errMsg := "Server NodePort whose externalTrafficPolicy is Local should be able to be connected from pod "
+	getHostNameUrl := fmt.Sprintf("%s/%s", url, "hostname")
+	getClientIPUrl := fmt.Sprintf("%s/%s", url, "clientip")
+
+	output, _, err := data.runCommandFromPod(testNamespace, clientCp, busyboxContainerName, []string{"wget", "-O", "-", getHostNameUrl, "-T", "1"})
+	require.NoError(t, err, errMsg)
+	require.Contains(t, output, testPodHostnameCp, fmt.Sprintf("hostname should be %s", testPodHostnameCp))
+
+	output, _, err = data.runCommandFromPod(testNamespace, clientCp, busyboxContainerName, []string{"wget", "-O", "-", getClientIPUrl, "-T", "1"})
+	require.NoError(t, err, errMsg)
+	require.Contains(t, output, clientCpIP, fmt.Sprintf("client IP should be %s", clientCpIP))
+
+	output, _, err = data.runCommandFromPod(testNamespace, clientWk, busyboxContainerName, []string{"wget", "-O", "-", getHostNameUrl, "-T", "1"})
+	require.NoError(t, err, errMsg)
+	require.Contains(t, output, testPodHostnameWk, fmt.Sprintf("hostname should be %s", testPodHostnameWk))
+
+	output, _, err = data.runCommandFromPod(testNamespace, clientWk, busyboxContainerName, []string{"wget", "-O", "-", getClientIPUrl, "-T", "1"})
+	require.NoError(t, err, errMsg)
+	require.Contains(t, output, clientWkIP, fmt.Sprintf("client IP should be %s", clientWkIP))
```

Comment:
Half of the code in this function is redundant as mentioned above and can be improved by calling the function twice with different input

Reply:
Updated.